[2024-12-22T13:54:42.530+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-22T13:54:42.600+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:54:36.600785+00:00 [queued]>
[2024-12-22T13:54:42.615+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:54:36.600785+00:00 [queued]>
[2024-12-22T13:54:42.616+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-22T13:54:42.650+0000] {taskinstance.py:2889} INFO - Executing <Task(DockerOperator): Notebook_job> on 2024-12-22 13:54:36.600785+00:00
[2024-12-22T13:54:42.669+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=383) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-12-22T13:54:42.678+0000] {standard_task_runner.py:72} INFO - Started process 384 to run task
[2024-12-22T13:54:42.681+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'docker_dag', 'Notebook_job', 'manual__2024-12-22T13:54:36.600785+00:00', '--job-id', '158', '--raw', '--subdir', 'DAGS_FOLDER/A_dag_test.py', '--cfg-path', '/tmp/tmpmbind4lm']
[2024-12-22T13:54:42.683+0000] {standard_task_runner.py:105} INFO - Job 158: Subtask Notebook_job
[2024-12-22T13:54:42.787+0000] {task_command.py:467} INFO - Running <TaskInstance: docker_dag.Notebook_job manual__2024-12-22T13:54:36.600785+00:00 [running]> on host 49ab4fbc6be0
[2024-12-22T13:54:42.890+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='docker_dag' AIRFLOW_CTX_TASK_ID='Notebook_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-22T13:54:36.600785+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-22T13:54:36.600785+00:00'
[2024-12-22T13:54:42.892+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-12-22T13:54:42.892+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-12-22T13:54:42.893+0000] {logging_mixin.py:190} INFO - Current task name:Notebook_job state:running start_date:2024-12-22 13:54:42.601205+00:00
[2024-12-22T13:54:42.893+0000] {logging_mixin.py:190} INFO - Dag name:docker_dag and current dag run status:running
[2024-12-22T13:54:42.893+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-22T13:54:43.078+0000] {docker.py:379} INFO - Starting docker container from image spark-cluster:version-1.0.0
[2024-12-22T13:54:44.285+0000] {docker.py:73} INFO - Input Notebook:  /home/jovyan/Notebooks/NB_Calendar.ipynb
[2024-12-22T13:54:44.292+0000] {docker.py:73} INFO - Output Notebook: /home/jovyan/Notebooks/output-NB_Calendar.ipynb
[2024-12-22T13:54:44.344+0000] {docker.py:73} INFO - Executing:   0%|          | 0/14 [00:00<?, ?cell/s]
[2024-12-22T13:54:45.172+0000] {docker.py:73} INFO - Executing notebook with kernel: python3
[2024-12-22T13:54:46.741+0000] {docker.py:73} INFO - Executing:   7%|▋         | 1/14 [00:02<00:31,  2.39s/cell]
[2024-12-22T13:54:48.241+0000] {docker.py:73} INFO - :: loading settings :: url = jar:file:/usr/local/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-12-22T13:54:48.286+0000] {docker.py:73} INFO - Ivy Default Cache set to: /home/jovyan/.ivy2/cache
[2024-12-22T13:54:48.287+0000] {docker.py:73} INFO - The jars for the packages stored in: /home/jovyan/.ivy2/jars
[2024-12-22T13:54:48.289+0000] {docker.py:73} INFO - io.delta#delta-spark_2.12 added as a dependency
[2024-12-22T13:54:48.289+0000] {docker.py:73} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-f45c8d4c-e10b-4d79-871f-71cc5562c2c8;1.0
[2024-12-22T13:54:48.290+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-22T13:54:50.882+0000] {docker.py:73} INFO - 	found io.delta#delta-spark_2.12;3.1.0 in central
[2024-12-22T13:54:51.378+0000] {docker.py:73} INFO - 	found io.delta#delta-storage;3.1.0 in central
[2024-12-22T13:54:54.570+0000] {docker.py:73} INFO - 	found org.antlr#antlr4-runtime;4.9.3 in central
[2024-12-22T13:54:54.860+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/io/delta/delta-spark_2.12/3.1.0/delta-spark_2.12-3.1.0.jar ...
[2024-12-22T13:54:56.630+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] io.delta#delta-spark_2.12;3.1.0!delta-spark_2.12.jar (2038ms)
[2024-12-22T13:54:56.941+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/io/delta/delta-storage/3.1.0/delta-storage-3.1.0.jar ...
[2024-12-22T13:54:57.160+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] io.delta#delta-storage;3.1.0!delta-storage.jar (529ms)
[2024-12-22T13:54:57.376+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/org/antlr/antlr4-runtime/4.9.3/antlr4-runtime-4.9.3.jar ...
[2024-12-22T13:54:57.603+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] org.antlr#antlr4-runtime;4.9.3!antlr4-runtime.jar (441ms)
[2024-12-22T13:54:57.605+0000] {docker.py:73} INFO - :: resolution report :: resolve 6299ms :: artifacts dl 3016ms
[2024-12-22T13:54:57.606+0000] {docker.py:73} INFO - 	:: modules in use:
[2024-12-22T13:54:57.606+0000] {docker.py:73} INFO - 	io.delta#delta-spark_2.12;3.1.0 from central in [default]
[2024-12-22T13:54:57.606+0000] {docker.py:73} INFO - 	io.delta#delta-storage;3.1.0 from central in [default]
[2024-12-22T13:54:57.607+0000] {docker.py:73} INFO - 	org.antlr#antlr4-runtime;4.9.3 from central in [default]
[2024-12-22T13:54:57.607+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:54:57.607+0000] {docker.py:73} INFO - 	|                  |            modules            ||   artifacts   |
[2024-12-22T13:54:57.607+0000] {docker.py:73} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-12-22T13:54:57.608+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:54:57.608+0000] {docker.py:73} INFO - 	|      default     |   3   |   3   |   3   |   0   ||   3   |   3   |
[2024-12-22T13:54:57.608+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-22T13:54:57.613+0000] {docker.py:73} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-f45c8d4c-e10b-4d79-871f-71cc5562c2c8
[2024-12-22T13:54:57.614+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-22T13:54:57.638+0000] {docker.py:73} INFO - 	3 artifacts copied, 0 already retrieved (5727kB/23ms)
[2024-12-22T13:54:57.779+0000] {docker.py:73} INFO - 24/12/22 13:54:57 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-22T13:54:57.877+0000] {docker.py:73} INFO - Setting default log level to "WARN".
[2024-12-22T13:54:57.877+0000] {docker.py:73} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-12-22T13:54:58.467+0000] {docker.py:73} INFO - 24/12/22 13:54:58 WARN StandaloneSchedulerBackend: Dynamic allocation enabled without spark.executor.cores explicitly set, you may get more executors allocated than expected. It's recommended to set spark.executor.cores explicitly. Please check SPARK-30299 for more details.
[2024-12-22T13:54:58.508+0000] {docker.py:73} INFO - 24/12/22 13:54:58 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:54:58.508+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:54:58.508+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:54:58.509+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:54:58.510+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:54:58.511+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:54:58.512+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:54:58.513+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:54:58.514+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:54:58.515+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:54:58.515+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:54:58.515+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:55:18.556+0000] {docker.py:73} INFO - 24/12/22 13:55:18 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:55:18.563+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:55:18.564+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:55:18.564+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:55:18.564+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:55:18.565+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:55:18.565+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:55:18.565+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:55:18.565+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:55:18.567+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:55:18.568+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:55:18.568+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:55:18.569+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:55:18.569+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:55:18.569+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:55:18.570+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:55:18.572+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:55:18.572+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:55:18.573+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:55:18.573+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:55:18.574+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:55:18.575+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:55:18.588+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:55:18.590+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:55:18.596+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:55:18.627+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:55:18.632+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:55:18.635+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:55:18.638+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:55:18.642+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:55:18.686+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:55:18.689+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:55:18.692+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:55:18.706+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:55:18.714+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:55:18.740+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:55:18.790+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:55:18.791+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:55:18.802+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:55:18.806+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:55:18.808+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:55:18.809+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:55:18.810+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:55:18.811+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:55:18.812+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:55:18.812+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:55:18.813+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:55:18.814+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:55:18.825+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:55:18.826+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:55:18.827+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:55:18.827+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:55:18.828+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:55:18.828+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:55:18.829+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:55:18.829+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:55:18.829+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:55:18.830+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:55:18.831+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:55:18.831+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:55:38.532+0000] {docker.py:73} INFO - 24/12/22 13:55:38 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2024-12-22T13:55:38.538+0000] {docker.py:73} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult: 
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2024-12-22T13:55:38.541+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/<unresolved>:7077
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - 	... 4 more
[2024-12-22T13:55:38.542+0000] {docker.py:73} INFO - Caused by: java.net.UnknownHostException: spark-master
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:801)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.net.InetAddress.getByName(InetAddress.java:1256)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
[2024-12-22T13:55:38.543+0000] {docker.py:73} INFO - 	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2024-12-22T13:55:38.544+0000] {docker.py:73} INFO - 	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2024-12-22T13:55:38.544+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2024-12-22T13:55:38.544+0000] {docker.py:73} INFO - 	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2024-12-22T13:55:38.544+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2024-12-22T13:55:38.544+0000] {docker.py:73} INFO - 	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175)
[2024-12-22T13:55:38.545+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
[2024-12-22T13:55:38.547+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
[2024-12-22T13:55:38.547+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
[2024-12-22T13:55:38.547+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
[2024-12-22T13:55:38.547+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[2024-12-22T13:55:38.548+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[2024-12-22T13:55:38.549+0000] {docker.py:73} INFO - 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[2024-12-22T13:55:38.549+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2024-12-22T13:55:38.549+0000] {docker.py:73} INFO - 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2024-12-22T13:55:38.549+0000] {docker.py:73} INFO - 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2024-12-22T13:55:38.549+0000] {docker.py:73} INFO - 	... 1 more
[2024-12-22T13:55:58.524+0000] {docker.py:73} INFO - 24/12/22 13:55:58 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
[2024-12-22T13:55:58.539+0000] {docker.py:73} INFO - 24/12/22 13:55:58 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
[2024-12-22T13:55:58.660+0000] {docker.py:73} INFO - 24/12/22 13:55:58 WARN Utils: spark.executor.instances less than spark.dynamicAllocation.minExecutors is invalid, ignoring its setting, please update your configs.
[2024-12-22T13:55:58.681+0000] {docker.py:73} INFO - 24/12/22 13:55:58 WARN StandaloneAppClient: Attempted to request executors before driver fully initialized.
[2024-12-22T13:55:59.146+0000] {docker.py:73} INFO - 24/12/22 13:55:59 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
[2024-12-22T13:55:59.731+0000] {docker.py:73} INFO - Executing:  21%|██▏       | 3/14 [01:15<05:04, 27.65s/cell]
[2024-12-22T13:56:02.368+0000] {docker.py:73} INFO - Executing:  21%|██▏       | 3/14 [01:18<04:46, 26.01s/cell]
[2024-12-22T13:56:02.377+0000] {docker.py:73} INFO - Traceback (most recent call last):
[2024-12-22T13:56:02.378+0000] {docker.py:73} INFO -   File "/opt/conda/bin/papermill", line 8, in <module>
[2024-12-22T13:56:02.380+0000] {docker.py:73} INFO -     sys.exit(papermill())
[2024-12-22T13:56:02.381+0000] {docker.py:73} INFO -              ^^^^^^^^^^^
[2024-12-22T13:56:02.381+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
[2024-12-22T13:56:02.381+0000] {docker.py:73} INFO -     return self.main(*args, **kwargs)
[2024-12-22T13:56:02.381+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:56:02.381+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1078, in main
[2024-12-22T13:56:02.382+0000] {docker.py:73} INFO -     rv = self.invoke(ctx)
[2024-12-22T13:56:02.382+0000] {docker.py:73} INFO -          ^^^^^^^^^^^^^^^^
[2024-12-22T13:56:02.385+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
[2024-12-22T13:56:02.385+0000] {docker.py:73} INFO -     return ctx.invoke(self.callback, **ctx.params)
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 783, in invoke
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -     return __callback(*args, **kwargs)
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -     return f(get_current_context(), *args, **kwargs)
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-22T13:56:02.386+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/cli.py", line 235, in papermill
[2024-12-22T13:56:02.387+0000] {docker.py:73} INFO -     execute_notebook(
[2024-12-22T13:56:02.387+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 131, in execute_notebook
[2024-12-22T13:56:02.387+0000] {docker.py:73} INFO -     raise_for_execution_errors(nb, output_path)
[2024-12-22T13:56:02.387+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 251, in raise_for_execution_errors
[2024-12-22T13:56:02.387+0000] {docker.py:73} INFO -     raise error
[2024-12-22T13:56:02.388+0000] {docker.py:73} INFO - papermill.exceptions.PapermillExecutionError: 
[2024-12-22T13:56:02.388+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-22T13:56:02.388+0000] {docker.py:73} INFO - Exception encountered at "In [3]":
[2024-12-22T13:56:02.388+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-22T13:56:02.389+0000] {docker.py:73} INFO - Py4JJavaError                             Traceback (most recent call last)
[2024-12-22T13:56:02.389+0000] {docker.py:73} INFO - Cell In[3], line 2
[2024-12-22T13:56:02.389+0000] {docker.py:73} INFO -       1 # Configure Spark with Delta Lake (if needed)
[2024-12-22T13:56:02.389+0000] {docker.py:73} INFO - ----> 2 spark = configure_spark_with_delta_pip(builder).getOrCreate()
[2024-12-22T13:56:02.390+0000] {docker.py:73} INFO -       4 driver_version = sys.version_info
[2024-12-22T13:56:02.391+0000] {docker.py:73} INFO -       5 worker_version = spark.sparkContext.pythonExec
[2024-12-22T13:56:02.391+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.392+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/pyspark/sql/session.py:500, in SparkSession.Builder.getOrCreate(self)
[2024-12-22T13:56:02.392+0000] {docker.py:73} INFO -     497     sc = SparkContext.getOrCreate(sparkConf)
[2024-12-22T13:56:02.392+0000] {docker.py:73} INFO -     498     # Do not update `SparkConf` for existing `SparkContext`, as it's shared
[2024-12-22T13:56:02.392+0000] {docker.py:73} INFO -     499     # by all sessions.
[2024-12-22T13:56:02.396+0000] {docker.py:73} INFO - --> 500     session = SparkSession(sc, options=self._options)
[2024-12-22T13:56:02.396+0000] {docker.py:73} INFO -     501 else:
[2024-12-22T13:56:02.396+0000] {docker.py:73} INFO -     502     getattr(
[2024-12-22T13:56:02.396+0000] {docker.py:73} INFO -     503         getattr(session._jvm, "SparkSession$"), "MODULE$"
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO -     504     ).applyModifiableSettings(session._jsparkSession, self._options)
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/pyspark/sql/session.py:589, in SparkSession.__init__(self, sparkContext, jsparkSession, options)
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO -     585         getattr(getattr(self._jvm, "SparkSession$"), "MODULE$").applyModifiableSettings(
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO -     586             jsparkSession, options
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO -     587         )
[2024-12-22T13:56:02.397+0000] {docker.py:73} INFO -     588     else:
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO - --> 589         jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO -     590 else:
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO -     591     getattr(getattr(self._jvm, "SparkSession$"), "MODULE$").applyModifiableSettings(
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO -     592         jsparkSession, options
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO -     593     )
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.398+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/java_gateway.py:1587, in JavaClass.__call__(self, *args)
[2024-12-22T13:56:02.399+0000] {docker.py:73} INFO -    1581 command = proto.CONSTRUCTOR_COMMAND_NAME +\
[2024-12-22T13:56:02.399+0000] {docker.py:73} INFO -    1582     self._command_header +\
[2024-12-22T13:56:02.399+0000] {docker.py:73} INFO -    1583     args_command +\
[2024-12-22T13:56:02.399+0000] {docker.py:73} INFO -    1584     proto.END_COMMAND_PART
[2024-12-22T13:56:02.399+0000] {docker.py:73} INFO -    1586 answer = self._gateway_client.send_command(command)
[2024-12-22T13:56:02.401+0000] {docker.py:73} INFO - -> 1587 return_value = get_return_value(
[2024-12-22T13:56:02.401+0000] {docker.py:73} INFO -    1588     answer, self._gateway_client, None, self._fqn)
[2024-12-22T13:56:02.401+0000] {docker.py:73} INFO -    1590 for temp_arg in temp_args:
[2024-12-22T13:56:02.402+0000] {docker.py:73} INFO -    1591     if hasattr(temp_arg, "_detach"):
[2024-12-22T13:56:02.402+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.402+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/protocol.py:326, in get_return_value(answer, gateway_client, target_id, name)
[2024-12-22T13:56:02.402+0000] {docker.py:73} INFO -     324 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2024-12-22T13:56:02.402+0000] {docker.py:73} INFO -     325 if answer[1] == REFERENCE_TYPE:
[2024-12-22T13:56:02.403+0000] {docker.py:73} INFO - --> 326     raise Py4JJavaError(
[2024-12-22T13:56:02.403+0000] {docker.py:73} INFO -     327         "An error occurred while calling {0}{1}{2}.\n".
[2024-12-22T13:56:02.404+0000] {docker.py:73} INFO -     328         format(target_id, ".", name), value)
[2024-12-22T13:56:02.404+0000] {docker.py:73} INFO -     329 else:
[2024-12-22T13:56:02.404+0000] {docker.py:73} INFO -     330     raise Py4JError(
[2024-12-22T13:56:02.404+0000] {docker.py:73} INFO -     331         "An error occurred while calling {0}{1}{2}. Trace:\n{3}\n".
[2024-12-22T13:56:02.407+0000] {docker.py:73} INFO -     332         format(target_id, ".", name, value))
[2024-12-22T13:56:02.408+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.408+0000] {docker.py:73} INFO - Py4JJavaError: An error occurred while calling None.org.apache.spark.sql.SparkSession.
[2024-12-22T13:56:02.408+0000] {docker.py:73} INFO - : java.lang.IllegalStateException: Cannot call methods on a stopped SparkContext.
[2024-12-22T13:56:02.409+0000] {docker.py:73} INFO - This stopped SparkContext was created at:
[2024-12-22T13:56:02.410+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.410+0000] {docker.py:73} INFO - org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2024-12-22T13:56:02.410+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-12-22T13:56:02.410+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-12-22T13:56:02.411+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-12-22T13:56:02.411+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-12-22T13:56:02.413+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-12-22T13:56:02.413+0000] {docker.py:73} INFO - py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-12-22T13:56:02.413+0000] {docker.py:73} INFO - py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-22T13:56:02.413+0000] {docker.py:73} INFO - py4j.Gateway.invoke(Gateway.java:238)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - The currently active SparkContext was created at:
[2024-12-22T13:56:02.414+0000] {docker.py:73} INFO - 
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-12-22T13:56:02.415+0000] {docker.py:73} INFO - py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-22T13:56:02.416+0000] {docker.py:73} INFO - py4j.Gateway.invoke(Gateway.java:238)
[2024-12-22T13:56:02.417+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-12-22T13:56:02.417+0000] {docker.py:73} INFO - py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-12-22T13:56:02.417+0000] {docker.py:73} INFO - py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-22T13:56:02.417+0000] {docker.py:73} INFO - py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-22T13:56:02.417+0000] {docker.py:73} INFO - java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO -          
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at org.apache.spark.SparkContext.assertNotStopped(SparkContext.scala:122)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:113)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:106)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-12-22T13:56:02.418+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-12-22T13:56:02.420+0000] {docker.py:73} INFO - 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-12-22T13:56:02.420+0000] {docker.py:73} INFO - 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-12-22T13:56:02.420+0000] {docker.py:73} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-12-22T13:56:02.421+0000] {docker.py:73} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-22T13:56:02.421+0000] {docker.py:73} INFO - 	at py4j.Gateway.invoke(Gateway.java:238)
[2024-12-22T13:56:02.421+0000] {docker.py:73} INFO - 	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-12-22T13:56:02.421+0000] {docker.py:73} INFO - 	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-12-22T13:56:02.421+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-22T13:56:02.422+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-22T13:56:02.423+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-22T13:56:02.986+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-22T13:56:03.007+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-12-22T13:56:03.008+0000] {logging_mixin.py:190} INFO - Task start:2024-12-22 13:54:42.601205+00:00 end:2024-12-22 13:56:03.000920+00:00 duration:80.399715
[2024-12-22T13:56:03.009+0000] {logging_mixin.py:190} INFO - Task:<Task(DockerOperator): Notebook_job> dag:<DAG: docker_dag> dagrun:<DagRun docker_dag @ 2024-12-22 13:54:36.600785+00:00: manual__2024-12-22T13:54:36.600785+00:00, state:running, queued_at: 2024-12-22 13:54:36.628382+00:00. externally triggered: True>
[2024-12-22T13:56:03.010+0000] {logging_mixin.py:190} INFO - Failure caused by Docker container failed: {'StatusCode': 1}
[2024-12-22T13:56:03.011+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=docker_dag, task_id=Notebook_job, run_id=manual__2024-12-22T13:54:36.600785+00:00, execution_date=20241222T135436, start_date=20241222T135442, end_date=20241222T135603
[2024-12-22T13:56:03.061+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-22T13:56:03.062+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 158 for task Notebook_job (Docker container failed: {'StatusCode': 1}; 384)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-22T13:56:03.125+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-22T13:56:03.163+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
